java/node/src/main/java/org/svarm/node/engine/impl/V1SingleEntryEngine.java

```diff
@@ -52,28 +56,27 @@ public class V1SingleEntryEngine implements TableDefinitionEngine {
   private static final String INTEGER_TYPE = "INTEGER";
   private static final String STRING_TYPE = "STRING";
   private final Metrics metrics;
-  private final SqlEngine sqlEngine;
+  private final TenantTableDataSourceManager dataSourceManager;
   private final JsonEngine jsonEngine;
 
   /**
    * Default constructor.
    *
-   * @param metrics    for analytics.
-   * @param sqlEngine  to execute sql.NO
-   * @param jsonEngine for managing json.
+   * @param metrics           for analytics.
+   * @param dataSourceManager for retrieving data sources of tenant dbs
+   * @param jsonEngine        for managing json.
    */
   @Inject
   public V1SingleEntryEngine(final Metrics metrics,
-                             final SqlEngine sqlEngine,
+                             final TenantTableDataSourceManager dataSourceManager,
                              final JsonEngine jsonEngine) {
+    this.dataSourceManager = dataSourceManager;
     this.jsonEngine = jsonEngine;
-    LOGGER.info("V1SingleEntryEngine({},{})", metrics, sqlEngine);
     this.metrics = metrics;
-    this.sqlEngine = sqlEngine;
   }
 
   /**
-   * Uses the murmur3/32b hashing function.
+   * Returns a hash function supplier, currently the murmur3/32b hashing function.
```

Comment:
Need a new story.. I moved the hashing function to a common class and standardized on murmur3. No need for table-specific implementations. (No change here, just making a new issue for this)

Reply:
https://github.com/wolpert/svarm/issues/43